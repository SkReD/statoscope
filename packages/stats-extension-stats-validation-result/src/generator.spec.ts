import { ExtensionDescriptor } from '@statoscope/stats/spec/extension';
import Generator from './generator';

const adapter: ExtensionDescriptor = { name: 'test-adapter', version: '7.7.7' };

jest.mock('../package.json', () => ({
  ...jest.requireActual('../package.json'),
  version: '8.8.8',
}));

test('should work', () => {
  const generator = new Generator(adapter);

  generator.handleRule('foo-rule', {
    description: 'foo-description',
    package: {
      name: 'foo-package',
      version: '1.0.0',
      homepage: 'https://statoscope.tech',
    },
  });

  generator.handleEntry(
    'foo-rule',
    {
      message: 'foo-message',
      compilation: 'foo-compilation',
      related: [
        { type: 'module', id: 'foo-foo-module' },
        { type: 'entry', id: 'foo-foo-entry' },
      ],
    },
    'warn',
  );

  generator.handleEntry('foo-rule', {
    message: 'bar-message',
    compilation: 'foo-compilation',
    related: [
      { type: 'module', id: 'foo-foo-module' },
      { type: 'module', id: 'foo-bar-module' },
      { type: 'entry', id: 'foo-foo-entry' },
      { type: 'entry', id: 'foo-bar-entry' },
    ],
  });

  generator.handleEntry('foo-rule', {
    message: 'bar-message',
    compilation: 'bar-compilation',
    related: [
      { type: 'module', id: 'bar-foo-module' },
      { type: 'module', id: 'bar-bar-module' },
      { type: 'entry', id: 'bar-foo-entry' },
      { type: 'entry', id: 'bar-bar-entry' },
    ],
  });

  expect(generator.get()).toMatchSnapshot();
});
