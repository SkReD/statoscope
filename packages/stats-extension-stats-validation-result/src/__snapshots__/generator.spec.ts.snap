// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
{
  "descriptor": {
    "adapter": {
      "name": "test-adapter",
      "version": "7.7.7",
    },
    "author": {
      "name": "Sergey Melyukov",
      "url": "https://twitter.com/smelukov",
    },
    "description": "Statoscope extension that collects stats validation result",
    "homepage": "https://github.com/statoscope/statoscope/",
    "name": "@statoscope/stats-extension-stats-validation-result",
    "version": "8.8.8",
  },
  "payload": {
    "compilations": [
      {
        "id": "foo-compilation",
        "items": [
          {
            "details": [],
            "id": 0,
            "message": "foo-message",
            "related": [
              {
                "id": "foo-foo-module",
                "type": "module",
              },
              {
                "id": "foo-foo-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "warn",
          },
          {
            "details": [],
            "id": 1,
            "message": "bar-message",
            "related": [
              {
                "id": "foo-foo-module",
                "type": "module",
              },
              {
                "id": "foo-bar-module",
                "type": "module",
              },
              {
                "id": "foo-foo-entry",
                "type": "entry",
              },
              {
                "id": "foo-bar-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "error",
          },
        ],
      },
      {
        "id": "bar-compilation",
        "items": [
          {
            "details": [],
            "id": 2,
            "message": "bar-message",
            "related": [
              {
                "id": "bar-foo-module",
                "type": "module",
              },
              {
                "id": "bar-bar-module",
                "type": "module",
              },
              {
                "id": "bar-foo-entry",
                "type": "entry",
              },
              {
                "id": "bar-bar-entry",
                "type": "entry",
              },
            ],
            "rule": "foo-rule",
            "type": "error",
          },
        ],
      },
    ],
    "rules": [
      {
        "descriptor": {
          "description": "foo-description",
          "package": {
            "homepage": "https://statoscope.tech",
            "name": "foo-package",
            "version": "1.0.0",
          },
        },
        "name": "foo-rule",
      },
    ],
  },
}
`;
